### DISCUSSION TITLE
[UX] Cycle items and paste with single hotkey

### DISCUSSION THREAD

**Discussion by gerardbosch** [hluk/CopyQ#1893] [Ideas]
Hi there! The goal of the following would be to improve the usability and it is inspired by Ditto (a Windows clipboard manager).

*Preconditions:*
- You enabled a hotkey to open the CopyQ history window in CopyQ settings. To move on with the example, let's say the hotkey is `Ctrl-;` on English US keyboard (could be the one of your choice).

## General idea

You want to paste, not the item at the tip the history, but one of the items at the top-most (say the second or third item).

Currently you have to do a lot of keystrokes just to paste the second/third/... item:
- `Ctrl-;` - which opens the history
- `↓`..., `⏎` - move down (n times), paste

which means moving the hands a lot from the home row and involves a lot of fingers, just to paste something recent.

In Ditto, you can do the following just to achieve the equivalent result, **only** involving the hotkey (`Ctrl-;` in the example):

- `hold Ctrl`, `;`, `;`, `release Ctrl` - Will open the history window, cycle to the **second** item on the second `;` keypress, and just paste it, closing the history, on `Ctrl` release.
- `hold Ctrl`, `;`, `;`, `;`, `release Ctrl` - Cycle to the **third** item and paste it.

:heart_eyes: You can quickly paste recent items by `Ctrl-;`...`;`

(Of course if you want to paste the first item, you don't need the history and just do `Ctrl-v` :smile:)

### Use case example 1: Sequential copy/paste

- You sequentially copy 3 strings (snippets/whatever) to your clipboard.
- Then, you want to paste it in the same order they were copied.
- You do:
  - `Ctrl-;;;` - to paste the first copied item/the third in the stack (once third item is pasted it becomes the first in the history)
  - `Ctrl-;;;` - to paste the second copied item (now the third in the stack)
  - `Ctrl-;;;` - to paste the third copied item (idem)

Very convenient, as you can repeat the same pattern `;;;`, and don't even need to think which position the item you want takes (you copied 3 items, you just issue 3 `;`).

### Use case example 2: Just paste the second item

Here you're sure only need to paste the second (or whatever) item, you do:
- `Ctrl-;;` - done! (or `;;;` or whatever :)

---

N.B. For a more complex history browsing, you can still do the usual `Ctrl-;` to open the history, browse with the arrows, and paste with return | or use the search, which is great :)

**Comment by hluk** [comment]
See: https://github.com/hluk/CopyQ/issues/1948#issuecomment-1087368505

**Reply by gerardbosch** [reply]
Thanks for your support, it totally worked! :D
